### PR TITLE
add http task timeout &  test unit

### PR DESCRIPTION
**Pull request by BoYiZhang**
#2882
image

### CODE REVIEW

**Comment by break60** [comment]
+1

**Comment by Eights-Li** [comment]
Please resolve the conflict

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/task/HttpParametersTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        Assert.assertEquals(HttpMethod.GET,httpParameters.getHttpMethod());
+        Assert.assertEquals(HttpCheckCondition.STATUS_CODE_DEFAULT,httpParameters.getHttpCheckCondition());
+        Assert.assertEquals("",httpParameters.getCondition());
+
+    }
+
+
+    @Test
+    public void testCheckParameters(){
+        String data1 = "{\"localParams\":[],\"httpParams\":[],\"url\":\"https://www.baidu.com/\"," +
```
Comment (Eights-Li) [inline dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/task/HttpParametersTest.java:58]: Use meaningful variable names， not data1, data2

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/task/HttpParametersTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+/**
+ * http parameter
+ */
+public class HttpParametersTest  {
+
+
+    @Test
+    public void testGenerator(){
+        String data1 = "{\"localParams\":[],\"httpParams\":[],\"url\":\"https://www.baidu.com/\"," +
```
Comment (Eights-Li) [inline dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/task/HttpParametersTest.java:41]: Use meaningful variable names， not data1, data2

File: dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/worker/task/http/HttpTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    @Test
+    public void testCheckParameters() {
+        Assert.assertTrue(httpTask.getParameters().checkParameters());
+    }
+
+
+    @Test
+    public void testGenerator(){
+        String data1 = "{\"localParams\":[],\"httpParams\":[],\"url\":\"https://github.com/\"," +
```
Comment (Eights-Li) [inline dolphinscheduler-server/src/test/java/org/apache/dolphinscheduler/server/worker/task/http/HttpTaskTest.java:124]: Use meaningful variable names， not data1, data2

**Comment by Eights-Li** [comment]
Please resolve the conflict then it will merge into dev

**Pull request closed without merging**